src/slices/users.slice.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+type Users = {
+  users: User[] | null;
+};
+
+const initialState: Users = {
+  users: null,
+};
+
+export const fetchAllUsers = createAsyncThunk<User[], void>(
+  'users/fetchAllUsers',
+  async () => {
+    try {
+      const result = await getUsers();
+
+      return result;
+    } catch (error) {
+      throw error;
+    }
+  },
+);
+
+const usersSlice = createSlice({
+  name: 'usersList',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(
+      fetchAllUsers.fulfilled,
+      (state, action: PayloadAction<User[]>) => {
+        state.users = action.payload;
+      },
+    );
+  },
+});
+
+export const {} = usersSlice.actions;
```

Comment:
The export statement for usersSlice.actions is exporting an empty object. If there are no actions to export, you can remove this line. Otherwise, ensure that any necessary actions are defined in the reducers object.